### PR TITLE
Refactor user location annotation (squash of #1222)

### PR DESCRIPTION
**Pull request by friedbunny**
Adds:
- Accuracy ring
- Heading indicator

Fixes:
- Pulse sync
- Visual appearance (removes slight aliasing)

Closes #1222 (squashed)

/cc @mapbox/mobile

### CODE REVIEW

**Comment by friedbunny** [comment]
Failing test is on Linux and related to headless image quality:

```
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Headless/HeadlessTest.render/40, where GetParam() = "raster-contrast"
```

File: platform/ios/MGLUserLocationAnnotationView.m [hunk 1]
```diff
@@ -48,164 +55,285 @@ - (instancetype)initWithCoder:(NSCoder *)decoder
...
             
-            [_haloLayer addAnimation:boundsAnimation forKey:@"animateScale"];
+            _oldHeadingAccuracy = self.annotation.heading.headingAccuracy;
+        }
+        
+        // update accuracy ring
+        //
+        if (_accuracyRingLayer)
+        {
+            // FIX: This stops EVERYTHING... and that isn't necessarily necessary, now is it?
```
Comment (1ec5) [inline platform/ios/MGLUserLocationAnnotationView.m]: Bailing here makes the code flow a bit harder to follow now. Can you look into whether it’s really necessary?

Comment (friedbunny) [inline platform/ios/MGLUserLocationAnnotationView.m]: I moved the zoom and heading accuracy is-necessary checks into the larger `if` statement in df2c1f5, which should be less jarring:

```diff
diff --git a/platform/ios/MGLUserLocationAnnotationView.m b/platform/ios/MGLUserLocationAnnotationView.m
index a6310a8..536326d 100644
--- a/platform/ios/MGLUserLocationAnnotationView.m
+++ b/platform/ios/MGLUserLocationAnnotationView.m
@@ -119,16 +119,10 @@ const CGFloat MGLUserLocationAnnotationHaloSize = 115.0;
             _oldHeadingAccuracy = self.annotation.heading.headingAccuracy;
         }
         
-        // update accuracy ring
+        // update accuracy ring (if zoom or horizontal accuracy have changed)
         //
-        if (_accuracyRingLayer)
+        if (_accuracyRingLayer && (_oldZoom != self.mapView.zoomLevel || _oldHorizontalAccuracy != self.annotation.location.horizontalAccuracy))
         {
-            // FIX: This stops EVERYTHING... and that isn't necessarily necessary, now is it?
-            if (_oldZoom == self.mapView.zoomLevel && _oldHorizontalAccuracy == self.annotation.location.horizontalAccuracy)
-            {
-                return;
-            }
-            
             CGFloat accuracyRingSize = [self calculateAccuracyRingSize];
             
             // only show the accuracy ring if it won't be obscured by the location dot
```

**Comment by 1ec5** [comment]
Looks good! Thanks for your persistence on this feature!

**Pull request merged**